---
title: "Predicting phenotypic and genotypic evolution given fluctuating selection"
author: "Esdras T"
format: 
  revealjs:
    standalone: true
    self-contained: true
    slide-number: true
    show-slide-number: all
editor: visual
---

## Aims

```{r, echo=FALSE, message=FALSE}
library(tidyverse)
library(stringr)
library(cowplot)
theme_set(theme_cowplot())
```

```{r, echo=FALSE, message=FALSE}
mytheme <- function(){
  theme_set(theme_cowplot())+
    theme(axis.title = element_text(face = "bold"),
          axis.text = element_text(face = "italic"),
          plot.title = element_text(hjust = 0.5),
          legend.position = "none")
}
```

-   Develop a polygenic model that accurately predicts genotypic and phenotypic evolution featuring fluctuating selection

## Genome

![](/home/etb68/YeastProj.dir/evogen-sims/ET_Yeast/Presentations/GenomeSLiM_Image.png){fig-align="center" height="80"}


## Genome

![](/home/etb68/YeastProj.dir/evogen-sims/ET_Yeast/Presentations/GenomeSLiM_Image.png){fig-align="center" height="80"}


-   Phenotype is the sum of all alleles' effect plus random noise 

## Parameters

WF model with 5 replicates each having the following parameters:

-   Heritability: 0.1, 0.5, and 0.8


## Parameters

WF model with 5 replicates each having the following parameters:

-   Heritability: 0.1, 0.5, and 0.8
-   Loci numbers: 1, 10, 70, 100, 300

## Parameters

WF model with 5 replicates each having the following parameters:

-   Heritability: 0.1, 0.5, and 0.8
-   Loci numbers: 1, 10, 70, 100, 300
-   Standard deviations: 1, 2, 3, 4

## Parameters

WF model with 5 replicates each having the following parameters:

-   Heritability: 0.1, 0.5, and 0.8
-   Loci numbers: 1, 10, 70, 100, 300
-   Standard deviations: 1, 2, 3, 4
-   Generation interval (GI): 10, 20, 30 (Linear changes)

## Parameters

WF model with 5 replicates each having the following parameters:

-   Heritability: 0.1, 0.5, and 0.8
-   Loci numbers: 1, 10, 70, 100, 300
-   Standard deviations: 1, 2, 3, 4
-   Generation interval (GI): 10, 20, 30 (Linear changes)
-   GI: sinusoidal1 (20) and sinusoidal2 (5, 10, 15, 30)

## Models

-   Neutral selection

-   Constant selection

-   Fluctuating selection(Linear & Sinusoidal)

## Fitness Function


$$
\text{Fitness} = \exp\left(-\frac{1}{{\text{phenotype}}} \times (\text{phenotype} - \text{optimum})^2\right)
$$

-   Optimum = mean(phenotype) ± nSD

## Fitness Function

$$
\text{Fitness} = \exp\left(-\frac{1}{{\text{phenotype}}} \times (\text{phenotype} - \text{optimum})^2\right)
$$

-   Optimum = mean(phenotype) ± nSD
```{r,fig.width=8, fig.height=6, echo=FALSE}
mu_pheno = 10
Opti_Gen <- tibble(Generation = seq(1:200),
                   Optimum = rep(c(rep(mu_pheno+4, 25), rep(mu_pheno-4, 25)),4))

Opt_Data <- Opti_Gen %>% 
  filter(Generation == 0 | Generation %% 25 == 0)

p_opt <- Opt_Data %>% 
  ggplot(aes(x = Generation, y = Optimum))+
  geom_line()+
  geom_hline(yintercept = mu_pheno, col = "darkred")+
  annotate("text", x = 100, y = mu_pheno, label = "Mean Phenotype", size = 3, vjust = -1.5, col = "darkred")+
  mytheme()
p_opt
```

## Genotype file

```{r, echo=FALSE}
mygeno <- read.csv("~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/LinFS.dir/genome1_n100_H0.5SD3Gen30.csv") %>% dplyr::select(-Origin) %>% filter(Position == 1043356)
knitr::kable(tail(mygeno, 50))
```

## Phenotype file

```{r, echo=FALSE}
mypheno <- read.csv("~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/LinFS.dir/MeanPhenotypes1_n100_H0.5SD3Gen30.csv")
knitr::kable(tail(mypheno, 50))
```

## Alle Frequency (single plot FS)

```{r , fig.width=8, fig.height=6, echo=FALSE}
mygeno %>% ggplot(aes(Generation, Frequency, color = as.character(Position)))+
  geom_line(linewidth = 0.2)+
  ggtitle("Alle freq plot h2 = 0.5, sd = 3 for 30 Generations")+
  mytheme()
```

## Allele Frequency (single model)

```{r, fig.width=8, fig.height=6, echo=FALSE}
source("~/YeastProj.dir/evogen-sims/ET_Yeast/Code/Selection_Models/WF.dir/Plotting/PrelimVisualSelect.R")
dir_path <- "~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/LinFS.dir/"
pattern <- "^genome.*0\\.5SD3Gen30\\.csv$"
ncol <- 3

plot_files(dir_path, pattern, ncol = ncol)
```

## Mean Allele Frequency (all variables)

```{r, fig.width=8, fig.height=6, echo=FALSE, message=FALSE}
source("~/YeastProj.dir/evogen-sims/ET_Yeast/Presentations/DataPhilesCode.R")
dir_path <- "~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/LinFS.dir/"
pattern <- "^genome\\d+_n100_H0[.5]+SD\\d+Gen\\d+\\.csv$"
output_dir <- "~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/Plots.dir/LinFS.dir/Frequency/"
allele_feq_fx(dir_path, pattern, output_dir)
```

## Phenotypes (single plot FS)

```{r, fig.width=8, fig.height=6, echo=FALSE}
mypheno %>% ggplot(aes(Generation, Phenotype))+
  geom_line(linewidth = 0.2)+
  ggtitle("Phenotype plot h2 = 0.5, sd = 3 for 30 Generations")+
  mytheme()
```

## Phenotypes (single model)

```{r, fig.width=8, fig.height=6, echo=FALSE, message=FALSE}
source("~/YeastProj.dir/evogen-sims/ET_Yeast/Code/Selection_Models/WF.dir/Plotting/PrelimVisualSelect.R")
dir_path <- "~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/LinFS.dir/"
pattern <- "^MeanPhenotypes.*0\\.5SD3Gen30\\.csv$"
ncol <- 3

Pheno_files(dir_path, pattern, ncol)
```

## Phenotypes (all variables)

```{r, fig.width=8, fig.height=6, echo=FALSE, message=FALSE}
source("~/YeastProj.dir/evogen-sims/ET_Yeast/Presentations/DataPhilesCode.R")
dir_path <- "~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/LinFS.dir/"
pattern <- "^MeanPhenotypes\\d+_n100+_H0[.5]+SD\\d+Gen\\d+\\.csv$"
output_dir <- "~/YeastProj.dir/evogen-sims/ET_Yeast/output.dir/Selection_Models/WF.dir/Plots.dir/LinFS.dir/Phenotypes/"
pheno_fx(dir_path, pattern, output_dir)

```

## Questions{.center}

-   How can I summarize the results to communicate parameters' effect given all simulations?
-   what I can change on the plots or which plots I should use to convey the information from simulation?
-   Any other statistical analyses, apart from spectral analysis should I perform?
-   Which biological questions should I ask?
